lib/Driver/UnixToolChains.cpp

```diff
@@ -337,13 +337,19 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
 }
 
 std::string toolchains::Android::getTargetForLinker() const {
+  assert(((getTriple().getArch() == llvm::Triple::arm &&
+           getTriple().getSubArch() == llvm::Triple::SubArchType::ARMSubArch_v7) ||
+          (getTriple().getArch() == llvm::Triple::aarch64 &&
+           getTriple().getSubArch() == llvm::Triple::SubArchType::ARMSubArch_v8)) &&
+         "Only arm targets are supported for Android");
```

Comment:
I'm not sure that this assertion is adding anything of value.

Reply:
I was just changing the existing assertion. Eventually the assertion will be come very difficult to read and maintain, so we might as well remove it now, if nobody disagrees.